### PR TITLE
Extensions to prop setting

### PR DESCRIPTION
**Pull request by gausie**
* Add increment and decrement functions for numeric property types
* Make set function return the set value (breaking change?)

### CODE REVIEW

**Review by horrible-little-slime — COMMENTED** [review]
I think you can use generic typing more rigorously using the PropertyType type, but also I think there's a reason we stopped doing that so maybe I'm wrong

**Comment by gausie** [comment]
Yes I think so too but can't remember why

File: src/property.ts [hunk 1]
```diff
@@ -219,34 +219,80 @@ export function get(property: string, _default?: unknown): unknown {
...
+ * @returns New value
+ */
+export function increment(
+  property: NumericProperty,
+  delta = 1,
+  max = Infinity
+) {
+  const value = get(property);
+
+  if (!isNumericProperty(property)) return value;
```
Comment (horrible-little-slime) [inline src/property.ts:269]: This feels unnecessary given that you already require `NumericProperty` typing

File: src/property.ts [hunk 2]
```diff
@@ -219,34 +219,80 @@ export function get(property: string, _default?: unknown): unknown {
...
+ * @returns New value
+ */
+export function decrement(
+  property: NumericProperty,
+  delta = 1,
+  min = Infinity
+) {
+  const value = get(property);
+
+  if (!isNumericProperty(property)) return value;
```
Comment (horrible-little-slime) [inline src/property.ts:291]: ditto

**Review by horrible-little-slime — APPROVED** [review]

**Comment by gausie** [comment]
I missed your review @horrible-little-slime! Thanks for merging. I think having explicit checks is good since libram is used by non-TS scripts, especially via `jsl`.

**Pull request merged**